119-deprecate-and-remove-kafka-spec-kafka-resources.md

```diff
@@ -0,0 +1,85 @@
+
+There are also other scenarios where users would use multiple node pools.
+For example, users might use node pools per availability zone to simplify node scheduling.
+In some of these scenarios, configuring resources through the `Kafka` CR might make more sense.
+But even in these cases, I would expect the controller node pool(s) to be there.
+So it would still not apply to all nodes, and users would need to use the resources in one or more Kafka node pools anyway.
+So removing the resource configuration from the `Kafka` CR does not seem to change things that much.
+
+## Proposal
+
+The `.spec.kafka.resources` section will be deprecated in the `v1beta2` API and will be removed in the `v1` CRD API.
+It will remain fully supported while the `v1beta2` API is supported and used (i.e., until Strimzi 0.52 / 1.0.0) and will work as before.
+
+### API Conversion
+
+The API conversion always converts only a single custom resource.
+So it cannot move the resource configuration from the `Kafka` CR to the `KafkaNodePool` CRs.
+It can either:
+* Remove the `.spec.kafka.resources` without any replacement
+* Fail the conversion and request the user to deal with the field manually (by removing it or moving it to the `KafkaNodePool` resources)
+
+Dropping the resources might cause significant problems for the Kafka clusters.
+This proposal therefore suggests failing the conversion and requesting users to deal with it manually.
+
+We should expect this to affect many users, because most clusters that existed before node pools had the resources configured in the `Kafka` CR as it was the only option at that time.
+So in some cases, even if the users have different configurations in the `KafkaNodePool` resources, they might still have the old settings in the `Kafka` CR as well.
+This means that this proposal might have a significant impact on our users.
+But there are other fields that will need to be converted manually, so this should be worth it if it helps us achieve a cleaner `v1` CRD API.
+
+### Implementation
+
+The proposal will be implemented in the `api` and `cluster-operator` modules (and in the not-yet-existing API Conversion Tool).
+The implementation will be trivial.
+In the first phase, as part of the 0.49.0 release:
+* The `.spec.kafka.resources` field will be deprecated and marked as present in the `v1beta2` API only.
+* The deprecation warnings will be suppressed in the `cluster-operator` module where it is used.
+* The API Conversion Tool will be updated to make sure it handles the conversion (raises the error).
+
+In the second phase, as part of the 0.52 / 1.0.0 release:
+* The `.spec.kafka.resources` field will be removed.
+* Its uses in the `cluster-operator` module will be removed as well.
+
+### Documentation
+
+The documentation will be updated with this change.
+We will make sure the regular documentation does not use or recommend the deprecated field.
+The conversion will be covered as part of the `v1` CRD API conversion documentation.
```

Comment:
```suggestion
The manual conversion will be covered as part of the `v1` CRD API conversion documentation.
```

I would just enforce that it's going to be manual, because the tool raises error and can't do much.